fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java

```diff
@@ -711,4 +717,118 @@ public String toString() {
     public boolean needAuditEncryption() {
         return !engineName.equals("olap");
     }
+
+    private void generatedColumnCheck(Analyzer analyzer) throws AnalysisException {
+        genreatedColumnCommonCheck();
+        Map<String, ColumnDef> nameToColumnDef = Maps.newHashMap();
+        for (ColumnDef columnDef : columnDefs) {
+            if (!columnDef.getGeneratedColumnInfo().isPresent()) {
+                nameToColumnDef.put(columnDef.getName(), columnDef);
+            }
+        }
+        List<GeneratedColumnUtil.ExprAndname> exprAndnames = Lists.newArrayList();
```

Comment:
mysql: Generated column can refer only to generated columns defined prior to it.
```sql
 create table tg2 (id int, c1 int as (c1), c2 int);
```